src/qibotn/backends/quimb.py

```diff
@@ -134,3 +145,143 @@ def execute_circuit(
+
+        This method takes a Qibo-style symbolic Hamiltonian (e.g., `X(0)*Z(1) + 2.0*Y(2)*Z(0)`)
+        as the observable, converts it into a Quimb observable and computes its expectation
+        value using the provided circuit. In case of multiple terms on the same group of qubits, they can be computed in a single contraction.
+        A grouping procedure is applied to optimize the number of contractions performed.
+
+        Args:
+            circuit: A Qibo quantum circuit object on which the expectation value
+                is computed.
+            observable: The observable whose expectation value we want to compute.
+                This must be provided in the symbolic Hamiltonian form supported by Qibo
+                (e.g., `X(0)*Y(1)` or `Z(0)*Z(1) + 1.5*Y(2)`).
+
+        Returns:
+            float: The expectation value (real part).
+        """
+
+        # Map the Qibo observable to Quimb operators and group local operators on the same sites
+        # for computing them in a single contraction. This does not work with CircuitMPS for some now
+        # for Quimb 1.11.1
+        operators_list, sites_list, coeffs_list = self._qiboobs_to_quimbobs(observable)
+        sites_list_grouped, operators_list_grouped, coeffs_list_grouped = (
+            self._group_by_tuples(sites_list, operators_list, coeffs_list)
+        )
+
+        if self.ansatz == "MPS":
+            if len(sites_list) - len(sites_list_grouped) > 10:
+                warnings.warn(
+                    "More than 10 local operators on the same sites are not being grouped as this is not compatible with CircuitMPS. Expected value computation can be more efficient without an MPS ansatz."
+                )
+            circ_ansatz = qtn.circuit.CircuitMPS
+            circ = circ_ansatz.from_openqasm2_str(circuit.to_qasm())
+            expectation_value = 0.0
+            for ops, sites, coeffs in zip(operators_list, sites_list, coeffs_list):
+                exp_values = circ.local_expectation(
+                    ops, where=sites, backend=self.backend, optimize=self.optimizer
+                )
+                expectation_value += np.dot(coeffs, exp_values)
+            return np.real(expectation_value)
+
+        else:
+            circ_ansatz = qtn.circuit.Circuit
+            circ = circ_ansatz.from_openqasm2_str(circuit.to_qasm())
+            expectation_value = 0.0
+            for ops, sites, coeffs in zip(
+                operators_list_grouped, sites_list_grouped, coeffs_list_grouped
+            ):
+                exp_values = circ.local_expectation(
+                    ops, where=sites, backend=self.backend, optimize=self.optimizer
+                )
+                expectation_value += np.dot(coeffs, exp_values)
+            return np.real(expectation_value)
+
+    @staticmethod
+    def _qiboobs_to_quimbobs(hamiltonian):
+        """
+        Convert a Qibo SymbolicHamiltonian into a Quimb-compatible decomposition.
```

Comment:
Specify here we are supporting only Pauli strings, which is what I understand from the code.